### ISSUE TITLE
Add `span` as a new data category to client reports

### ISSUE THREAD

**Issue by whatyouhide** [getsentry/sentry-elixir#812]
This needs #784 to be merged first.

**Comment by whatyouhide** [comment]
@solnic I've fallen a bit behind with changes here and changes to what we're reporting with the new log stuff and whatnot. Do you think this is still relevant?

**Comment by solnic** [comment]
Yes, we're not sending client reports for dropped transactions yet, and the SDKs are supposed to do it according to https://develop.sentry.dev/sdk/telemetry/client-reports/#span-outcomes
